application-tls-termination/README.md

```diff
@@ -0,0 +1,260 @@
+
+In this example, we will use Traefik, which comes with the k3d cluster.  It is a common ingress controller in its own right, and serves as a good extended example as configuring it to send TLS traffic to the back end application is not possible with the built in Ingress object.
+
+> Note: Other Ingress controllers will be differ in their configuration, but the [manifests](./manifests) will give a good example of the specific information about paths, ports and service names.
+
+
+## Setup Steps
+
+### Prerequisites
+
+Use the [basic helm install guide](../helm-basic/README.md) to install Howso Platform, and ensure it is running correctly. See the [TLDR](../common/README.md#basic-helm-install) for a quick start.
+
+Install the [step](https://smallstep.com/docs/step-cli/) certificate tool. 
+
+### Configure Howso Platform TLS sidecars
+
+The values file for the howso-platform chart will need to turn on the podTLS feature ..
+```yaml
+podTLS:
+  enabled: true 
+```
+... and globally turn Ingress object creation off.
+
+```yaml
+ingress:
+  enabled: false
+```
+
+From your installation of the [helm basic example](../helm-basic/README.md), update the chart with these [manifests](./manifests/howso-platform.yaml) additions. 
+
+```sh
+helm upgrade howso-platform oci://registry.how.so/howso-platform/stable/howso-platform --namespace howso --values helm-basic/manifests/howso-platform.yaml --values custom-ingress/manifests/howso-platform.yaml 
+```
+
+### Creating sidecar server Certificates
+
+Without further intervention, some of the Howso Platform pods will no longer be able to start as they expect to find secrets containing certificates that don't yet exist.
+
+In the next steps we'll create certificates for a root CA and each required service using the `step` CLI tool, then create the corresponding Kubernetes secrets.
+
+> Note: This example uses a root CA so that the Ingress controller just needs to trust a single certificate.  There is an existing platform root CA secret (platform-ca) that is created by the platform-cert-generation jobs, which also creates certificates for use by the UMS to provide an Oauth authorization server, and a default ingress cert.  It is possible to extract and reuse this root ca, but this example will focus solely of the TLS termination certificates.
+
+
+#### Root CA
+
+First, create the root CA:
+
+```bash
+step certificate create root-ca root-ca.crt root-ca.key \
+    --profile root-ca \
+    --no-password --insecure
+```
+
+Create Root ca secret:
+```bash
+kubectl create secret generic platform-app-tls-ca \
+  --from-file=ca.crt=root-ca.crt \
+  -n howso
+```
+
+#### Platform PyPI Server
+
+Create the certificate for the PyPI server:
+
+```bash
+step certificate create platform-pypi.default.svc.cluster.local platform-pypi-server-tls.crt platform-pypi-server-tls.key \
```

Comment:
Are there a set of rules that describe the parameters you're using here?  ("I have my own custom PKI, what parameters do I need to pass into it?")

Reply:
fair - I've added a new section that covers this